chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/SpringBasedHttpClient.java

```diff
@@ -0,0 +1,161 @@
+        }
+
+        public String getValue() {
+            return value;
+        }
+    }
+
+    /**
+     * This field is a container for {@link RestTemplate} parameters which can be passed by the
+     * {@link SpringBasedHttpClient#SpringBasedHttpClient(Map)} constructor or by {@link JHttpQuery#clientParams} property.<p>
+     * Client parameters of query {@link JHttpQuery#clientParams} will override the same parameters of <b>clientParams</b> field
+     * for the time of execution of this query.
+     * <p><p>
+     * The list of client params (look at {@link JSpringBasedHttpClientParameters}): <p>
+     * - {@code Map<String, ?> defaultUriVariables} (look at {@link RestTemplate#setDefaultUriVariables(Map)}) <p>
+     * - {@code ResponseErrorHandler errorHandler} (look at {@link RestTemplate#setErrorHandler(ResponseErrorHandler)}) <p>
+     * - {@code List<HttpMessageConverter<?>> messageConverters} (look at {@link RestTemplate#setMessageConverters(List)}) <p>
+     * - {@code UriTemplateHandler uriTemplateHandler} (look at {@link RestTemplate#setUriTemplateHandler(UriTemplateHandler)}) <p>
+     * - {@code List<ClientHttpRequestInterceptor> interceptors} (look at {@link RestTemplate#setInterceptors(List)}) <p>
+     * - {@code ClientHttpRequestFactory requestFactory} (look at {@link RestTemplate#setRequestFactory(ClientHttpRequestFactory)}) <p>
+     */
+    private Map<String, Object> clientParams;
+
+    private RestTemplate restTemplate = new RestTemplate();
+
+    public SpringBasedHttpClient() { }
+
+    public SpringBasedHttpClient(Map<String, Object> clientParams) {
+        this.clientParams = clientParams;
+    }
+
+    @Override
+    public JHttpResponse execute(JHttpEndpoint endpoint, JHttpQuery query) {
+        setRestTemplateParamsForQuery(query);
+        URI endpointURI = endpoint.getURI(query.getQueryParams());
+
+        if (endpoint.getProtocol() == HTTPS) {
+            restTemplate.setRequestFactory(getSslAllTrustRequestFactory());
+        }
+
+        RequestEntity requestEntity = mapToRequestEntity(query, endpointURI);
+        ResponseEntity responseEntity = restTemplate.exchange(endpointURI, query.getMethod(), requestEntity, Object.class);
+
+        restoreDefaultRestTemplateParams();
+        return mapToJHttpResponse(responseEntity);
+    }
+
+    private void setRestTemplateParamsForQuery(JHttpQuery query) {
+        Map<String, Object> queryClientParams = new HashMap<>(query.getClientParams());
+        clientParams.entrySet().forEach(entry -> queryClientParams.putIfAbsent(entry.getKey(), entry.getValue()));
+        setRestTemplateParams(queryClientParams);
+    }
+
+    private void restoreDefaultRestTemplateParams() {
+        setRestTemplateParams(this.clientParams);
+    }
+
+    private void setRestTemplateParams(Map<String, Object> clientParams) {
+        clientParams.forEach((parameterKey, parameterVal) -> {
+            if (parameterKey.equals(DEFAULT_URI_VARIABLES.value)) {
```

Comment:
and here just get an action from that by parameterKey value and apply the action